dnsdist-1.8.x: move to ubuntu 22 for package building and ci

romeroalx opened the pull request:
### Short description
Fixes <https://github.com/PowerDNS/pdns/actions/runs/14784303728/job/41509789973> by moving to use `ubuntu-22.04` runners.

Also backported this PR: <https://github.com/PowerDNS/pdns/pull/13154> that includes moving from `nose` to `pytests`. This increases the number of commits in this PR.

### Checklist
<!-- please indicate if any of these things are done/included with this Pull Request. Not all boxes need to be checked for the Pull Request to be accepted -->
I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ] <!-- remove this line if your PR is against master --> checked that this code was merged to master

rgacogne commented:
Thanks a lot, Alexis! I have a slightly different proposal here: https://github.com/rgacogne/pdns/tree/dnsdist-1.8.x-ubuntu-22 I dropped the backport of the "Add metrics for health-check failures" commit which introduces changes that I was not planning on backporting, and added a new commit fixing the health-check tests instead. Let me know if that makes sense to you :)

romeroalx commented:
Superseded by #15556

romeroalx commented:
> Thanks a lot, Alexis! I have a slightly different proposal here: https://github.com/rgacogne/pdns/tree/dnsdist-1.8.x-ubuntu-22 I dropped the backport of the "Add metrics for health-check failures" commit which introduces changes that I was not planning on backporting, and added a new commit fixing the health-check tests instead. Let me know if that makes sense to you :)

Yes, makes complete sense. Just created a clean PR with your branch: #15556

Thank you!
